webapp/publisher/snaps/views.py

```diff
@@ -1344,36 +1348,89 @@ def get_snap_builds(snap_name):
-        owner="https://api.launchpad.net/devel/~build.snapcraft.io",
-        store_name=details["snap_name"],
-    )
+    context["github_user"] = github.get_user()
+    context["github_repositories"] = github.get_user_repositories()
+
+    # Get built snap in launchpad with this store name
+    lp_snap = launchpad.get_snap_by_store_name(details["snap_name"])
+
+    if lp_snap:
+        context["github_repository"] = get_github_repository(
+            lp_snap["git_repository_url"]
+        )
 
-    context["snap_builds_enabled"] = len(lp_snaps) >= 1
-    if context["snap_builds_enabled"]:
-        context["snap_builds"] = []
         bsi_url = flask.current_app.config["BSI_URL"]
 
-        for build in lp_snaps[0].builds:
-            link = build_link(bsi_url, lp_snaps[0], build)
+        builds = launchpad.get_collection_entries(
+            lp_snap["builds_collection_link"]
+        )
+
+        context["snap_builds_enabled"] = bool(builds)
+
+        for build in builds:
+            link = build_link(bsi_url, lp_snap, build)
             status = map_build_and_upload_states(
-                build.buildstate, build.store_upload_status
+                build["buildstate"], build["store_upload_status"]
             )
 
             context["snap_builds"].append(
                 {
-                    "id": build.self_link.split("/")[-1],
-                    "arch_tag": build.arch_tag,
-                    "datebuilt": build.datebuilt,
-                    "duration": build.duration,
+                    "id": build["self_link"].split("/")[-1],
+                    "arch_tag": build["arch_tag"],
+                    "datebuilt": build["datebuilt"],
+                    "duration": build["duration"],
                     "link": link,
-                    "logs": build.build_log_url,
-                    "revision_id": build.revision_id,
+                    "logs": build["build_log_url"],
+                    "revision_id": build["revision_id"],
                     "status": status,
-                    "title": build.title,
+                    "title": build["title"],
                 }
             )
 
     return flask.render_template("publisher/builds.html", **context)
+
+
+@publisher_snaps.route("/<snap_name>/builds", methods=["POST"])
+@login_required
+def get_snap_builds_post(snap_name):
+    try:
+        details = api.get_snap_info(snap_name, flask.session)
+    except ApiResponseErrorList as api_response_error_list:
+        if api_response_error_list.status_code == 404:
+            return flask.abort(404, "No snap named {}".format(snap_name))
+        else:
+            return _handle_error_list(api_response_error_list.errors)
+    except ApiError as api_error:
+        return _handle_errors(api_error)
+
+    github_repo = flask.request.form.get("github_repository")
+    git_url = "https://github.com/" + github_repo
+
+    # Get built snap in launchpad with this store name
+    launchpad = Launchpad()
```

Comment:
Can be removed and we just use the top level one?

Reply:
My bad...